core/src/main/java/org/lflang/generator/GeneratorBase.java

```diff
@@ -627,6 +629,24 @@ protected void cleanIfNeeded(LFGeneratorContext context) {
     }
   }
 
+  /** Return a {@code DockerGenerator} instance suitable for the target. */
+  protected abstract DockerGenerator getDockerGenerator(LFGeneratorContext context);
+
+  /** Create Dockerfiles and docker-compose.yml, build, and create a launcher. */
+  protected boolean buildUsingDocker() {
+    // Create docker file.
+    var dockerCompose = new DockerComposeGenerator(context);
+    var dockerData = getDockerGenerator(context).generateDockerData();
+    try {
+      dockerData.writeDockerFile();
+      dockerCompose.writeDockerComposeFile(List.of(dockerData));
+    } catch (IOException e) {
+      context.getErrorReporter().nowhere().error("Error while writing Docker files");
```

Comment:
Something like this might improve debuggability:
```suggestion
      context.getErrorReporter().nowhere().error("Error while writing Docker files: " + (e.getMessage() == null ? "No cause given" : e.getMessage()));
```